### ISSUE TITLE
CA1508 AvoidDeadConditionalCode - false positive on params argument

### ISSUE THREAD

**Issue by BrightLight** [dotnet/roslyn-analyzers#3845]
#### Analyzer package
[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version
v3.0.0 (Latest)

#### Diagnostic ID
[CA1508](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1508): Avoid dead conditional code

#### Repro steps
```csharp
protected void SetupControlHandlers(params IControlHandler[] controlHandlers)
{
  if (controlHandlers == null || controlHandlers.Length == 0)
  {
    return;
  }
  [...]
}
```

I get an issue reported for `controlHandlers == null` although `controlHandlers` can indeed be null ([proof](https://stackoverflow.com/questions/6584131/c-sharp-is-it-possible-to-have-null-params))

#### Expected behavior
No issue is reported.

#### Actual behavior
I get an issue reported for `controlHandlers == null`.

**Comment by neolution-ch** [comment]
This is clearly a bug and should be fixed: `params` can be `null`, so there is no dead conditional code here.

**Comment by Tan90909090** [comment]
Minimum reproducible example:
```cs
class C
{
    static void Main(params string[] args)
    {
        if (args == null) { return; } // CA1508 here
    }
}
```
